src/Hpack/Config.hs

```diff
@@ -275,18 +275,36 @@ instance FromJSON Condition where
 
 instance HasFieldNames Condition
 
-data ThenElse = ThenElse {
+data FlatThen a = FlatThen {
+  _flatThenCondition :: String
+, _flatThenBody :: CaptureUnknownFields (Section a a)
+} deriving (Eq, Show, Generic)
```

Comment:
What we do here with `FlatThen` is essentially saying: On this level we want to accept all the fields of `Condition` and all the fields of `Section`.  I added a `Product` data type that allows us to solve this kind of situation generically.  So instead of `FlatThen` we can have `Product (Section a a) Condition` now.